qiskit/transpiler/coupling.py

```diff
@@ -320,6 +320,56 @@ def from_grid(cls, num_rows, num_columns, bidirectional=True):
         )
         return cmap
 
+    @classmethod
+    def from_heavy_hex(cls, distance, bidirectional=True):
+        """Return a heavy hexagon graph coupling map
+
+        A heavy hexagon graph is described in:
+
+        https://journals.aps.org/prx/abstract/10.1103/PhysRevX.10.011022
+
+        Args:
+            distance (int): The code distance for the generated heavy hex
+                graph. The value for distance can be any odd positive integer.
+                The distance relates to the number of qubits by:
+                :math:`n = \\frac{5d^2 - 2d - 1}{2}` where :math:`n` is the
+                number of qubits and :math:`d` is the ``distance`` parameter.
```

Comment:
I'm wondering how one might make a coupling map like ibmq_mumbai using this, since it is a 27-qubit graph. Maybe if there was a way to remove some qubits from a coupling map, a user could start with a base and manually work towards that.
<img width="630" alt="image" src="https://user-images.githubusercontent.com/8622381/134228995-95448e3a-7d87-4a2a-b5f1-7823a312a374.png">

Reply:
Yeah, I think the only way right now would be to leverage the `reduce()` method after calling the heavy hex generator. Something like:

```python
from retworkx.visualization import mpl_draw
from qiskit.transpiler import CouplingMap

cmap = CouplingMap.from_heavy_hex(5)
mumbai_cmap = cmap.reduce([4, 40, 3, 39, 27, 2, 9, 44, 38, 1, 8, 26, 43, 7, 42, 29, 6, 13, 47, 41, 5, 12, 28, 46, 11, 45, 10])
mpl_draw(mumbai_cmap.graph, with_labels=True).show()
```
![Figure_1](https://user-images.githubusercontent.com/2447371/134234504-0ec3b582-9bda-4ffa-b637-c5838b748670.png)

It's not the prettiest interface for that though, mostly because the node order retworkx creates is by type (data, syndrome, and flag qubits are created up front based on the number given the distance: https://github.com/Qiskit/retworkx/blob/0.10.2/src/generators.rs#L1692-L1702

Reply:
With `bidirectional=False`:

![undirected](https://user-images.githubusercontent.com/2447371/134235893-14e7b6f7-237a-4e7a-aac5-c4dae43a1937.png)